pkgs/servers/mail/mailpit/default.nix

```diff
@@ -19,23 +20,45 @@ buildGoModule rec {
     hash = "sha256-jT9QE0ikp9cJlT8qtfPPjKOUuqWyQk94D3UbkyaGXa8=";
   };
 
-  vendorHash = "sha256-XBYIO7fdo5EahJB7EcAuY9SGKZb8dsvoJHp/D5LO5Qo=";
+  # Separate derivation, because if we mix this in buildGoModule, the separate
+  # go-modules build inherits specific attributes and fails. Getting that to
+  # work is hackier than just splitting the build.
+  ui = stdenv.mkDerivation {
+    pname = "mailpit-ui";
+    inherit src version;
+
+    npmDeps = fetchNpmDeps {
+      inherit src;
+      hash = "sha256-6VCs8125fTJkZW+eZgK56j7ccK8tcGhIXiq2HkYp4XM=";
+    };
 
-  npmDeps = fetchNpmDeps {
-    inherit src;
-    hash = "sha256-6VCs8125fTJkZW+eZgK56j7ccK8tcGhIXiq2HkYp4XM=";
+    nativeBuildInputs = [ nodejs python3 libtool npmHooks.npmConfigHook ];
+
+    buildPhase = ''
```

Comment:
Let's not forget the missing hooks in `buildPhase` and `installPhase`.

Reply:
There's not much use in this case, I think? The `ui` derivation is not accessible, I believe, so there's also no opportunity to manipulate it with `overrideAttrs` or so.

Reply:
You're right, it's just my habit to check for these automatically.